src/k2/httpproxy/HTTPProxy.cpp

```diff
@@ -347,10 +296,118 @@ HTTPProxy::_handleGetSchema(shd::GetSchemaRequest&& request) {
     });
 }
 
+void HTTPProxy::setQueryRecord(const sh::String& collectionName, shd::SKVRecord::Storage&& key, dto::SKVRecord& k2record) {
```

Comment:
This name is confusing to me as I'm not seeing how we're 'setting' a query record anywhere. I'm also not sure what a 'query record' is.
I think this is the equivalent for _shdRecToK2, but with storage. So something like shdStorageToK2Record would be more descriptive.